app/models/user.rb

```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  before_save { email.downcase! }
+  validates :name,  presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: { maximum: 255 }, format: { with: VALID_EMAIL_REGEX }, uniqueness: { case_sensitive: false }
+  has_secure_password
```

Comment:
how has_secure_password work?

Reply:
It's a Rails method with these functionality:
- we can save a securely hashed password_digest attribute to db.
- add a pair of virtual attributes password and password_confirmation, validates that the password is confirmed by password_confirmation, so you have to enter the password twice to confirm it.
- Ensures that password_digest is present before a model is saved.
- An authenticate method that returns the user when the password is correct (and false otherwise)